src/kixi/event_replay.clj

```diff
@@ -0,0 +1,176 @@
+(ns kixi.event-replay
+  (:gen-class)
+  (:require [aero.core :as aero]
+            [amazonica.aws.s3 :as s3]
+            [clj-time.core :as t]
+            [clj-time.format :as f]
+            [clj-time.periodic :as p]
+            [clojure.java.io :as io]
+            [clojure.spec.alpha :as s]
+            [clojure.string :as string]
+            [kixi.event-replay.inputstream-to-events :refer [input-stream->events]]))
+
+(def config (atom {}))
+
+(def datehour-formatter
+  (f/formatter :date-hour))
+
+(def parse-datahour
+  (partial f/parse datehour-formatter))
+
+(def unparse-datahour
+  (partial f/unparse datehour-formatter))
+
+(defn datehour?
+  [x]
+  (if (instance? org.joda.time.DateTime x)
+    x
+    (try
+      (if (string? x)
+        (parse-datahour x)
+        :clojure.spec.alpha/invalid)
+      (catch IllegalArgumentException e
+        :clojure.spec.alpha/invalid))))
+
+(def datehour
+  (s/conformer datehour? identity))
+
+(s/def ::start-datehour datehour)
+(s/def ::end-datehour 
+  (s/or :t datehour
+        :n nil?))
+
+(s/def ::target-stream string?)
+(s/def ::s3-base-dir
+  (s/and string?
+         #(string/ends-with? % "-backup")))
+
+(s/def ::config
+  (s/keys :req-un [::start-datehour
+                   ::end-datehour
+                   ::target-stream
+                   ::s3-base-dir]))
+
+(defn validate-config
+  [config]
+  (if (s/valid? ::config config)
+    [true (s/conform ::config config)]
+    [false (s/explain-data ::config config)]))
+
+(def one-hour (t/hours 1))
+
+(defn hour-sequence
+  []
+  (let [{:keys [start-datehour
+                end-datehour]} @config
+        end-datehour (or end-datehour 
+                         (unparse-datahour (t/now)))
+        start (parse-datahour start-datehour)
+        end (parse-datahour end-datehour)]    
+    (p/periodic-seq start
+                    (t/plus end
+                            one-hour)
+                    one-hour)))
+
+(defn hour->s3-prefix
+  [hour]
+  (->> [(t/year hour)       
```

Comment:
can't you use `clj-time.format/unparse` here?

Reply:
Could create and slash delimited format I guess, but wldnt gain too much.